src/StdCheats.sol

```diff
@@ -222,6 +233,84 @@ abstract contract StdCheatsSafe {
         assumeNotBlacklisted(token, addr);
     }
 
+    function assumeAddressIsNot(AddressType addressType, address addr) internal virtual {
+        if (addressType == AddressType.Payable) {
+            assumeNoPayable(addr);
+        } else if (addressType == AddressType.NonPayable) {
+            assumeNoNonPayable(addr);
+        } else if (addressType == AddressType.ZeroAddress) {
+            assumeNoZeroAddress(addr);
+        } else if (addressType == AddressType.Precompiles) {
+            assumeNoPrecompiles(addr);
+        } else if (addressType == AddressType.ForgeAddresses) {
+            assumeNoForgeAddresses(addr);
+        }
+    }
+
+    function assumeAddressIsNot(AddressType addressType1, AddressType addressType2, address addr) internal virtual {
+        assumeAddressIsNot(addressType1, addr);
+        assumeAddressIsNot(addressType2, addr);
+    }
+
+    function assumeAddressIsNot(
+        AddressType addressType1,
+        AddressType addressType2,
+        AddressType addressType3,
+        address addr
+    ) internal virtual {
+        assumeAddressIsNot(addressType1, addr);
+        assumeAddressIsNot(addressType2, addr);
+        assumeAddressIsNot(addressType3, addr);
+    }
+
+    function assumeAddressIsNot(
+        AddressType addressType1,
+        AddressType addressType2,
+        AddressType addressType3,
+        AddressType addressType4,
+        address addr
+    ) internal virtual {
+        assumeAddressIsNot(addressType1, addr);
+        assumeAddressIsNot(addressType2, addr);
+        assumeAddressIsNot(addressType3, addr);
+        assumeAddressIsNot(addressType4, addr);
+    }
+
+    function _isPayable(address addr) private returns (bool) {
+        uint256 size;
+        assembly {
+            size := extcodesize(addr)
+        }
+
+        if (size == 0) {
+            // return false if no code
+            return false;
+        } else {
+            require(addr.balance < UINT256_MAX, "balance exceeds max uint256");
+            uint256 origBalanceTest = address(this).balance;
+            uint256 origBalanceAddr = address(addr).balance;
+            (bool success,) = payable(addr).call{value: 1}("");
+
+            // reset balances
+            vm.deal(address(this), origBalanceTest);
+            vm.deal(addr, origBalanceAddr);
+
+            return success;
+        }
+    }
+
+    function assumeNoPayable(address addr) internal virtual {
```

Comment:
Hmm, wondering if we should use "No" or "Not" in each of these: `assumeNoPrecompiles` makes sense, but it could have instead been "assumeNotPrecompile". Maybe the rule of thumb is "No" for plural and "Not" for singular? By that logic `assumeNoPayable`, `assumeNoNonPayable, `assumeNoZeroAddress` should all be "Not". I'm not great at grammar, so maybe I'm overcomplicating this 😅

Maybe we should use `assumePayable` and `assumeNotPayable` for those two, instead of the double negative of `assumeNoNonPayable`?

Reply:
Changed them all to singular and `Not*`.

Reply:
One thing I did notice when adjusting the tests for `Payable`/`NotPayable` is that in the `forge test` context, the console address has no code while the vm has a fallback and sending 1 wei results in the payable check returns true.